test/controllers/images_controller_test.rb

```diff
@@ -96,14 +84,29 @@ def test_should_show_image
     get image_path(id: testimage.id)
     assert_response :ok
     assert_select "img[src='http://abc.png']", count: 1
-    assert_select '.tags-list', count: 0 # no tags block
+    assert_select '.js-card-tag', count: 0 # no tags block
+  end
+
+  def test_should_show_image_not_found
+    get image_path(id: '-1')
+    assert_redirected_to images_path
+    assert_equal 'Image was not found.', flash[:danger]
   end
 
-  def test_should_show_image_with_tag
-    testimage = Image.create!(imageurl: 'http://abc.png', tag_list: 'abc')
+  def test_should_show_image_with_tags_alphabetically
+    testimage = Image.create!(imageurl: 'http://abc.png', tag_list: 'c, d, b, a')
+    expected_names = %w[a b c d]
+
     get image_path(id: testimage.id)
     assert_response :ok
     assert_select "img[src='http://abc.png']", count: 1
-    assert_select '.tags-list', 'abc'
+
+    assert_select '.js-tags-list', count: 1 do
+      assert_select 'a', count: 4 do |buttons|
+        buttons.each_with_index do |_button, index|
+          assert_select 'a', expected_names[index]
+        end
+      end
+    end
```

Comment:
I think we're missing a couple of tests for the `#delete` action here :)

Reply:
@D-D-M , We've added a couple of controller tests for the `#delete` action, and are ready for a review!  Thanks!